lib/auth/auth.go

```diff
@@ -3993,6 +3993,10 @@ func (a *Server) CreateAccessRequestV2(ctx context.Context, req types.AccessRequ
 		req.SetRequestedResourceIDs(requestedResourceIDs)
 	}
 
+	if err := a.verifyAccessRequestMonthlyLimit(ctx); err != nil {
```

Comment:
This function must be called after the `DryRun()` we use the dry run option in the UI to calculate max duration. The current implementation will break it.
Note: This UI flow happens when you create a new request, which technically is blocked anyways when you're out of the limit, but I think it's safer to allow people to call this endpoint with a dry run rather than block it.